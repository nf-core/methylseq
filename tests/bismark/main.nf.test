import com.askimed.nf.test.util.FileUtil;

nextflow_pipeline {

    name "bismark"
    script "main.nf"

    test("Single End") {
        when {
            params {
                aligner = "bismark"
                save_reference = true
                outdir = "$outputDir"
            }
        }

        then {
            // Clean up anything from previous runs
            FileUtil.deleteDirectory(new File(".nf-test/indexes/"));
            new File(".nf-test/indexes/").mkdirs()
            // Copy the ref genome to a stable location for next tests
            FileUtil.copyDirectory(
                new File("$outputDir/bismark/reference_genome/BismarkIndex/").getAbsolutePath(),
                new File(".nf-test/indexes/BismarkIndex/").getAbsolutePath()
            )

            // Test results
            assert workflow.success
            assert snapshot(workflow.trace.tasks().size()).match()
            // FIXME assert snapshot(workflow).match()
            assert new File("$outputDir/multiqc/bismark/multiqc_report.html").exists()

            assert new File("$outputDir/bismark/methylation_calls/splitting_report/SRR389222_sub1_trimmed_bismark_bt2.deduplicated_splitting_report.txt").exists()
            assert new File("$outputDir/bismark/methylation_calls/splitting_report/SRR389222_sub2_trimmed_bismark_bt2.deduplicated_splitting_report.txt").exists()
            assert path("$outputDir/bismark/methylation_calls/splitting_report/SRR389222_sub2_trimmed_bismark_bt2.deduplicated_splitting_report.txt").md5 == "3038a90a6e513c3b1c0de540439eb101"
        }
    }

    test("Single End with index") {
        when {
            params {
                aligner = "bismark"
                // Generated by previous test
                bismark_index = ".nf-test/indexes/BismarkIndex/"
                cytosine_report = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.trace.tasks().size()).match()
            // FIXME assert snapshot(workflow).match()
        }
    }

    test("Single End RRBS") {
        when {
            params {
                aligner = "bismark"
                skip_trimming = true
                save_reference = true
                rrbs = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.trace.tasks().size()).match()
            // FIXME assert snapshot(workflow).match()
        }
    }

    test("Single End RRBS with index") {
        when {
            params {
                aligner = "bismark"
                skip_trimming = true
                rrbs = true
                save_reference = true
                // Generated by previous test
                bismark_index = ".nf-test/indexes/BismarkIndex/"
                cytosine_report = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.trace.tasks().size()).match()
            // FIXME assert snapshot(workflow).match()
        }
    }

    test("Single End NOMe-seq") {
        when {
            params {
                aligner = "bismark"
                nomeseq = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.trace.tasks().size()).match()
        }
    }

}
