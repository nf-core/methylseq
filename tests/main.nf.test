import com.askimed.nf.test.util.FileUtil;

nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "methylseq"

    test("METHYLSEQ Workflow main.nf") {

        when {
            params {
                save_reference = true
                outdir = "$outputDir"
            }
        }

        then {
            // Assert Workflow Success
            assert workflow.success

            // Bismark Methylation Calls Splitting Report
            assert snapshot(path("$outputDir/bismark/methylation_calls/splitting_report/Ecoli_10K_methylated_1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt"),
                            path("$outputDir/bismark/methylation_calls/splitting_report/SRR389222_sub1_trimmed_bismark_bt2.deduplicated_splitting_report.txt"),
                            path("$outputDir/bismark/methylation_calls/splitting_report/SRR389222_sub2_trimmed_bismark_bt2.deduplicated_splitting_report.txt")).match("splitting_report")

            // Bismark Summary Report
            assert snapshot(path("$outputDir/bismark/summary/bismark_summary_report.txt")).match("summary")

            // Reference Genome
            assert snapshot(path("$outputDir/bismark/reference_genome/BismarkIndex/genome.fa")).match("reference_genome")

            // Qualimap Report
            assert snapshot(path("$outputDir/qualimap/Ecoli_10K_methylated/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub1/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub2/genome_results.txt")).match("qualimap")

            // Software Versions
            assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions")
        }
    }

}

