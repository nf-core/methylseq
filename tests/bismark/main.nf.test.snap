{
    "single_end_nomeseq": {
        "content": [
            "Ecoli_10K_methylated_1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt:md5,292c73be0dfaad92ee7d54ec3e5d0dd6",
            "SRR389222_sub1_trimmed_bismark_bt2.deduplicated_splitting_report.txt:md5,d70ee229a29b68db8634fa7d568b7c53",
            "SRR389222_sub2_trimmed_bismark_bt2.deduplicated_splitting_report.txt:md5,3038a90a6e513c3b1c0de540439eb101",
            "genome_results.txt:md5,7886cc1a96ee892e49898b0d69ed66f2",
            "genome_results.txt:md5,79287f69ff6c8a308f0bb988c97f0e15",
            "genome_results.txt:md5,29e911b75d7160e964530e9607110164",
            "software_versions.yml:md5,f281f6db0fb109fd50575688149859e3"
        ],
        "timestamp": "2023-04-25T16:12:43+0000"
    },
    "single_end": {
        "content": [
            "Ecoli_10K_methylated_1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt:md5,292c73be0dfaad92ee7d54ec3e5d0dd6",
            "SRR389222_sub1_trimmed_bismark_bt2.deduplicated_splitting_report.txt:md5,d70ee229a29b68db8634fa7d568b7c53",
            "SRR389222_sub2_trimmed_bismark_bt2.deduplicated_splitting_report.txt:md5,3038a90a6e513c3b1c0de540439eb101",
            "genome_results.txt:md5,7886cc1a96ee892e49898b0d69ed66f2",
            "genome_results.txt:md5,79287f69ff6c8a308f0bb988c97f0e15",
            "genome_results.txt:md5,29e911b75d7160e964530e9607110164",
            "software_versions.yml:md5,f281f6db0fb109fd50575688149859e3"
        ],
        "timestamp": "2023-04-25T16:12:43+0000"
    },
    "single_end_rrbs": {
        "content": [
            "Ecoli_10K_methylated_R1_bismark_bt2_pe_splitting_report.txt:md5,9f72ee663677fb74fde3752b08350858",
            "SRR389222_sub1_bismark_bt2_splitting_report.txt:md5,f59eac7a30ec7f1f690aac921a5591c0",
            "SRR389222_sub2.merged_bismark_bt2_splitting_report.txt:md5,43ca6c76e47ce2eb1c8dd6027475b5a6",
            "genome_results.txt:md5,41becdcecee6ebbdd46f472629731892",
            "genome_results.txt:md5,7a86634fd1c768d8302e071088641815",
            "genome_results.txt:md5,9e9b6841f3b3a46a299c61c516aa125e",
            "software_versions.yml:md5,986af83c1bf68311b22432ed217161bd"
        ],
        "timestamp": "2023-04-25T16:12:43+0000"
    }
}